tensorflow_addons/losses/python/focal_loss.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.keras import backend as K
+
+
+def sigmoid_focal_crossentropy(y_true, y_pred, alpha=0.25,
+                               gamma=2.0,
+                               from_logits=False):
+  """Focal loss down-weights well classified examples and focuses on the hard
+  examples. See https://arxiv.org/pdf/1708.02002.pdf for the loss definition.
+
+  # Args
+      y_true: tensor of true targets.
+      y_pred: tensor of predicted targets.
+      alpha: balancing factor.
+      gamma: modulating factor.
+
+  # Returns:
+      loss: a float tensor of shape [batch_size, num_anchors, num_classes]
+              representing the value of the loss function.
+  """
+  assert gamma >= 0, "Value of gamma should be greater than or equal to zero"
```

Comment:
Can you make this raise a ValueError instead of an assertion? If you look at caveat 1 in this doc:
https://dbader.org/blog/python-assert-tutorial

You'll see that assertions actually get compiled into null ops and will not get checked at runtime (from .pyc files for example)